<a href="https://colab.research.google.com/github/minjunkim-99/pytohn/blob/main/py_bban.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**CH1 파이썬 문법**

## 자료형  
#### 숫자형  
* 정수 : 123, -20, 0  
* 실수 : 123.45, -423.24, 6.08e9  
* 8진수 : 0o456, 0o123  
* 16진수 : 0xFF, 0x0D, 0x0A

#### 문자열  
1.큰 따옴표 : "Hello World! it'\s"  
2.작은 따옴표 : '안녕하세요'  
3.큰 따옴표 3 : """Hello!"""  
4.작은 따옴표 4 : ''''대충 문장이 길면 작은 따옴표 4개 쓴다는 뜻''''  


#### 변수
* 문자 또는 밑줄로 시작 (alpha, beta, _kim)  
* 대소문자로 구분한다.(sum, Sum, SUM)
* 영문자, 숫자, 밑줄(a-z, 0-9,_)  
* 파이썬 키워드는 사용 불가

In [ ]:
myName="minjun Kim" #카멜 표기법 약칭 낙타표기법
my_Name="김민준" #스네이크 표기법
MyName="kim" #파스칼 표기법
_my_Name="korea"
MYNAME="failed"
my2name="12345"
#틀린 예시 (왠지 시험에 나올거같음)
#2myname='9876'
#my-name="6974"
#my name='00000'
myStr='123' # 작은 따옴표는 글자
myNum=123 # <--경우는 숫자
print(myStr,myNum)
print(type(myStr))
print(type(myNum))

123 123
<class 'str'>
<class 'int'>


####여러개 변수 할당

In [ ]:
x,y,z="바나나","알러지","원숭이"
print(x)
print(y)
print(z)

바나나
알러지
원숭이


In [ ]:
a=b=c="오렌지"
print(a)
print(b)
print(c)

오렌지
오렌지
오렌지


In [ ]:
fruit=["포도","사과","오렌지"]
x,y,z=fruit
print(x)
print(y)
print(z)

포도
사과
오렌지


In [ ]:
x="Life"
y="is"
z="Beautiful"
print(x,y,z)
print(x+y+z)

Life is Beautiful
LifeisBeautiful


In [ ]:
a=1
b=2
c=3
print(a,b,c,)
print(a+b+c)

1 2 3
6


#### 데이터 유형  
+ 텍스트  
+ 숫자  
+ 불(Bool)  

In [ ]:
a=10
b=3
c=a/b #나눗셈
d=a//b #몫
e=a%b #나머지
f=a*b #곱셈
g=a**b #제곱(시험에 나옴)
print(c,d,e,f,g)

3.3333333333333335 3 1 30 1000


####input() 함수를 이용한 계산기

In [ ]:
a = int(input("첫번째 숫자를 입력해주세요: "))
b = int(input("두번째 숫자를 입력해주세요: "))
result = a + b
print(a, '+', b, '=', result)
result = a - b
print(a, '-', b, '=', result)
result = a ** b
print(a, '**', b, '=', result)
result = a // b
print(a, '//', b, '=', result)

In [ ]:
num1 = input("숫자입력:")
num2 = input("숫자입력:")
result = num1 + num2
print(type(num1))
print(num1,'+', num2,'=',result)

#### 전화번호, 이름 무게를 입력받아 출력:
입력한 내용은 이름 "김민준",  
전화번호 "010-4237-1537" 입니다.  
무게는 입력한 g x 10원 = 산출 금액 출력

In [ ]:
a=input("전화번호를 입력해주세요 : ")
b=input("이름을 입력해주세요 : ")
c=int(input("무게를 입력해주세요 : "))
d= "전화번호 : "
e= "이름 : "
f= "산출 금액: "
g= "원"
result = c*10
print(d, a)
print(e, b)
print(f, result, g)

#CH2. print() 서식 출력

예시 %7.1에서 7은 .을 포함해서 총 7개의 자릿수를 가지며 1은 소수점의 자릿수이다.  
%07.1일때 123.45는 00123.4가 된다.
(시험 출제)

In [ ]:
print("%d" % 123)  
print("%5d" % 123)
print("%05d" % 123)
print("%f" % 123.45)
print("%7.1f" % 123.45)
print("%7.3f" % 123.45)

print("%s" % "대한민국")
print("%8s" % "대한민국")

### Format 함수 : 순서 지정 가능

In [ ]:
print("{0:d} {1:5d} {2:05d}".format(123,456,789))

In [ ]:
print("{2:d} {1:5d} {0:05d}".format(123,456,789))

서식 출력

In [ ]:
print("\n줄바꿈\n연습입니다.")

In [ ]:
print("\t탭키\t연습입니다.")

In [ ]:
print("글자가 \"강조\"되는 효과1")
print("글자가 \'강조\'되는 효과2")
print("역슬래시 3개출력 \\\\\\")
print(r"\n \t \" \\ \'@를 그대로 출력") #그대로 출력됨

#### 관계 연산자

In [ ]:
a,b = 10,20
print(a==b, a!=b, a>b, a<b, a>=b, a<=b)

논리 연산자

In [ ]:
a=99
print((a>100) and (a<200))
print((a>100) or (a<200))
print(not(a==100))

In [ ]:
if(123):
  print("진실이면 보입니다.")
if(0):
  print("거짓이면 안 보입니다.")

### 퀴즈2 숫자 입력을 받아 동전으로 바꿔주는 프로그램( 500원, 100원, 50원, 10원)

In [ ]:
bill, c500, c100, c50, c10 = 0,0,0,0,0
bill=int(input("값을 입력해주세요. : "))
c500 = bill // 500
bill %= 500 #bill %= 500
c100 = bill // 100 
bill %= 100
c50 = bill // 50
bill %=  50
c10 = bill // 10
bill %= 10
print("500원 %d 개" % c500)
print("100원 %d 개" % c100)
print("50원 %d 개" % c50)
print("10원 %d 개" % c10)
print("잔돈 %d원" % bill)

#### Week 2

## 조건문

In [ ]:
a=90
if(a<100):
  print("100보다 작음")
  print("거짓이면 이문장은 보이지 않음")
print("프로그램 끝")

In [ ]:
a=101
if(a<100):
  print("100보다 작음")
else :
  print("100보다 크다")
print("프로그램 끝")

In [ ]:
a=int(input("정수를 입력하세요:"))
if(a%2==0):
  print("짝수")
else:
  print("홀수")

In [ ]:
a=111
if(a>50):
  if(a<100):
    print("50<a<100")
  else:
    print("a>100")
else:
  print("a<50")

In [ ]:
fruit=['사과','배','감','포도']
fruit.append('딸기')
if '딸기' in fruit :
  print("딸기가 있음")
print(fruit)

In [ ]:
import random
number=[]
for num in range(0,7):
  number.append(random.randrange(1,49))
print("생성된 리스트", number)

###반복문

In [ ]:
for i in range(0,3,1): #시작값 , 끝값+1, 증가값 (시험 출제 확률 있음)
  print("%d 안녕하세요" % i)  #반복 내용
  #print(i)
print("행복하세요")    #

0 안녕하세요
1 안녕하세요
2 안녕하세요
행복하세요


In [ ]:
for i in range(1,9,1):
  print("%d" % i, end=" ")
print("\n")
for i in range(1,9,1):
  print("%d" % i, end="")

In [ ]:
sum=0
for i in range(1,11,1):
  sum = sum + i
print("1~10까지 합계 : %d" % sum)

#### 퀴즈3 : 숫자 입력을 받아 1부터 입력된 숫자까지의 합을 for문을 사용하여 출력하시오.

In [ ]:
sum, a = 0, 0
a=int(input("정수를 입력하세요:"))
for i in range(1,a+1,1):
  sum = sum + i
print("1~n 까지의 합계 : %d" % sum)

#### 중첩 for문

In [ ]:
a,b=0,0
for a in range(1,10,1):
  for b in range(1,10,1):
    print("%d x %d = %2d \t " % (b,a,a*b), end='')
  print("")
print("") 
c,d=0,0
for c in range(1,10,1):
  for d in range(6,10,1):
    print("%d x %d = %2d \t " % (d,c,c*d), end='')
  print("")

1 x 1 =  1 	 2 x 1 =  2 	 3 x 1 =  3 	 4 x 1 =  4 	 5 x 1 =  5 	 6 x 1 =  6 	 7 x 1 =  7 	 8 x 1 =  8 	 9 x 1 =  9 	 
1 x 2 =  2 	 2 x 2 =  4 	 3 x 2 =  6 	 4 x 2 =  8 	 5 x 2 = 10 	 6 x 2 = 12 	 7 x 2 = 14 	 8 x 2 = 16 	 9 x 2 = 18 	 
1 x 3 =  3 	 2 x 3 =  6 	 3 x 3 =  9 	 4 x 3 = 12 	 5 x 3 = 15 	 6 x 3 = 18 	 7 x 3 = 21 	 8 x 3 = 24 	 9 x 3 = 27 	 
1 x 4 =  4 	 2 x 4 =  8 	 3 x 4 = 12 	 4 x 4 = 16 	 5 x 4 = 20 	 6 x 4 = 24 	 7 x 4 = 28 	 8 x 4 = 32 	 9 x 4 = 36 	 
1 x 5 =  5 	 2 x 5 = 10 	 3 x 5 = 15 	 4 x 5 = 20 	 5 x 5 = 25 	 6 x 5 = 30 	 7 x 5 = 35 	 8 x 5 = 40 	 9 x 5 = 45 	 
1 x 6 =  6 	 2 x 6 = 12 	 3 x 6 = 18 	 4 x 6 = 24 	 5 x 6 = 30 	 6 x 6 = 36 	 7 x 6 = 42 	 8 x 6 = 48 	 9 x 6 = 54 	 
1 x 7 =  7 	 2 x 7 = 14 	 3 x 7 = 21 	 4 x 7 = 28 	 5 x 7 = 35 	 6 x 7 = 42 	 7 x 7 = 49 	 8 x 7 = 56 	 9 x 7 = 63 	 
1 x 8 =  8 	 2 x 8 = 16 	 3 x 8 = 24 	 4 x 8 = 32 	 5 x 8 = 40 	 6 x 8 = 48 	 7 x 8 = 56 	 8 x 8 = 64 	 9 x 8 = 72 	 
1 x 9 =  9 	 2 x 9 = 18 	 3 x 9 = 27 	 4 x 9 = 36 	 5 x 

#### Week 3  

#CH3. 자료형  
## 1. 인덱스와 슬라이싱

## 리스트는 대괄호로 정의  
index는 0부터 시작함  
1=[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]  
print(1[0])  
1  
시험 출제 예정  
print(1[0:4])
[1, 2, 3, 4]

In [ ]:
L=[0,1,2,3,4,5,6,7,8,9,10]
print(L[1])
print(L[0:4]) #미만으로 출력된다.
print(L[0:9:3]) #9까지의 값을 0부터 +3해서 출력

1
[0, 1, 2, 3]
[0, 3, 6]
11


In [ ]:
L=[0,1,2,3,4,5,6,7,8,9,10]
print(len(L))
print(L[len(L)-1])

11
10


In [ ]:
L=[0,1,2,3,4,5,6,7,8,9,10]
L[0]=99
L[9]='가나다'
L[1]=[1,2,3]
print(L)
print(L[9])

[99, [1, 2, 3], 2, 3, 4, 5, 6, 7, 8, '가나다', 10]
가나다
[1, 2, 3]


In [ ]:
a=[1,2,3]
b=[4,5,6]
c=a+b
print(a+b)
print(c)
print(a*3) #를 3번 출력

[1, 2, 3, 4, 5, 6]
[1, 2, 3, 4, 5, 6]
[1, 2, 3, 1, 2, 3, 1, 2, 3]


In [ ]:
L=[1,2,3,4,5]
print(L)
L.append(6) #L에 6을 추가하라는 명령어 append로 추가할 시 리스트의 마지막에 추가됨
print(L)
L.remove(3) # 해당 요소값을 삭제하는 명령어
print(L)

[1, 2, 3, 4, 5]
[1, 2, 3, 4, 5, 6]
[1, 2, 4, 5, 6]


In [ ]:
k=['a','b','c','d']
k.remove('d')
print(k)

['a', 'b', 'c']


In [ ]:
k="Guns Germs and Steel!"
print(k[:3])
print(k[5:])
print(k[-10:-1])

Gun
Germs and Steel!
and Steel


In [ ]:
k=" God is love!"
print(k.upper()) # 대문자
print(k.lower()) # 소문자
print(k.strip()) # 띄어쓰기 공간 제거

 GOD IS LOVE!
 god is love!
God is love!


In [ ]:
a=" God, is, love!"
print(a.split(",")) # 구분

[' God', ' is', ' love!']


In [ ]:
a=["apple","banana","cherry"]
a.append("orange")
print(a)

['apple', 'banana', 'cherry', 'orange']


In [ ]:
a =["apple","banana","cherry"]
a.insert(1,"orange") #.insert:지정된 곳에 요소값 삽입
print(a)

['apple', 'orange', 'banana', 'cherry']


In [ ]:
a =["apple","banana","cherry"]
a.remove("banana") # .remove:지정한 요소값 삭제
print(a)

['apple', 'cherry']


In [ ]:
a =["apple","banana","cherry"]
a.pop() # .pop은 지정없이 적용하면 마지막 요소값이 제거됨
print(a)
a.pop(1) # 지정한 위치인 banana가 사라짐
print(a)

['apple', 'banana']
['apple']


In [ ]:
a =["apple","banana","cherry"]
del a[0] #지정 삭제 del a 는 전제 삭제
print(a)
a.clear() #del a
print(a)

['banana', 'cherry']
[]


In [ ]:
fruit=["apple","banana","cherry"]
for x in fruit:
  print(x)

apple
banana
cherry


In [ ]:
fruit=["apple","banana","cherry"]
for i in range(len(fruit)):
  print(fruit)

['apple', 'banana', 'cherry']
['apple', 'banana', 'cherry']
['apple', 'banana', 'cherry']


#### SORT

In [ ]:
a=[5,3,6,8,1,9,0]
a.sort() # 순서대로 정렬
print(a)

[0, 1, 3, 5, 6, 8, 9]


In [ ]:
a=[5,3,6,8,1,9,0,2,4,7]
a.sort(reverse=True) # 역정렬
print(a)

[9, 8, 7, 6, 5, 4, 3, 2, 1, 0]


In [ ]:
fruit=["apple","banana","Kiwi","cherry","orange"]
fruit.sort() # 대소문자 구분하여 정렬
print(fruit)
fruit.sort(key=str.lower) # 대소문자 구분 없이 정렬
print(fruit)
fruit.reverse()
print(fruit) #역정렬

['Kiwi', 'apple', 'banana', 'cherry', 'orange']
['apple', 'banana', 'cherry', 'Kiwi', 'orange']
['orange', 'Kiwi', 'cherry', 'banana', 'apple']


In [ ]:
fruit=["apple","banana","Kiwi","cherry","orange"]
myList=fruit.copy()
print(myList)
cpList=list(fruit)
print(cpList)

['apple', 'banana', 'Kiwi', 'cherry', 'orange']
['apple', 'banana', 'Kiwi', 'cherry', 'orange']


## 튜플 자료형 Tuple   
리스트와 비슷하지만 값을 변경할 수 없음  
(시험 출제 예정)

In [ ]:
l=[1,2,3]
t=(4,5,6) #튜플은 ()를 사용한다. (시험 출제가능성 높음)
l[0]=5
print(l)
print(t)

[5, 2, 3]
(4, 5, 6)


In [ ]:
f=("apple","banana","Kiwi","cherry","orange")
print(len(f))

5


In [ ]:
t=("apple",) # 1가지 함수를 튜플로 인식시키기 위해서는 ,가 있어야 한다.
print(t)
f=("banana") # ,를 넣지 않았기에 튜플로 인식이 되지 않음
print(f)
print(type(t))
print(type(f))

('apple',)
banana
<class 'tuple'>
<class 'str'>


In [ ]:
f=("apple","banana","Kiwi","cherry","orange")
print(f[:4])
print(f[2:])

('apple', 'banana', 'Kiwi', 'cherry')
('Kiwi', 'cherry', 'orange')


In [ ]:
f=("apple","banana","Kiwi","cherry","Orange")
if "apple" in f:
  print("Yes, 'apple' is in")
else:
  print("No")

Yes, 'apple' is in


####list<->tuple

In [ ]:
firm=['Samsung','LG','SK']
tdata=tuple(firm)
print(firm)
print(tdata)

['Samsung', 'LG', 'SK']
('Samsung', 'LG', 'SK')


In [ ]:
t=("apple","banana","Kiwi","cherry")
y=list(t) 
y.append("Orange")
t=tuple(y) # 튜플을 list로 바꾼다음 Orange를 넣고 다시 튜플로 변환함(시험출제 가능성 높음)
print(t)

('apple', 'banana', 'Kiwi', 'cherry', 'Orange')


In [ ]:
t=("apple","banana","cherry")
q=("kiwi",)
t += q
print(t)

('apple', 'banana', 'cherry', 'kiwi')


In [ ]:
t=("apple","banana","Kiwi","cherry")
l=list(t)
l.remove("Kiwi")
t=tuple(l)
print(t)
del t # 튜플도 삭제 자체는 가능하다.
# print t

('apple', 'banana', 'cherry')


#### 사전(Dict) 자료형  
키값을 사용하고 중괄호{}를 사용한다.  


In [ ]:
d={
    'a':1,
    'b':2,
    'c':3
}
print(d)
print(d.keys())
print(d.values())
print(d.items())

{'a': 1, 'b': 2, 'c': 3}
dict_keys(['a', 'b', 'c'])
dict_values([1, 2, 3])
dict_items([('a', 1), ('b', 2), ('c', 3)])


In [ ]:
car={
    "brand":"BM",
    "model":"GT",
    "year" :"1988"
}
print(car)
print(len(car))
car["year"]=2000
car["item"]=123456
car.update({"color":"red"})
print(car)

{'brand': 'BM', 'model': 'GT', 'year': '1988'}
3
{'brand': 'BM', 'model': 'GT', 'year': 2000, 'item': 123456, 'color': 'red'}


In [ ]:
# 항목 제거
car={
    "brand":"BM",
    "model":"GT",
    "year" :"1988"
}
car.pop("model")
print(car)

{'brand': 'BM', 'year': '1988'}


In [ ]:
car={
    "brand":"BM",
    "model":"GT",
    "year" :"1988"
}
car.popitem()
print(car)

{'brand': 'BM', 'model': 'GT'}


In [ ]:
car={
    "brand":"BM",
    "model":"GT",
    "year" :"1988"
}
car.clear() # 비움
del car # 삭제

#### LOOP

In [ ]:
# 값을 하나씩 인쇄
car={
    "brand":"BM",
    "model":"GT",
    "year" :1988
}
for x in car:
  print(car[x])

BM
GT
1988


In [ ]:
car={
    "brand":"BM",
    "model":"GT",
    "year" :1988
}
for x in car.keys():
  print(x)

brand
model
year


In [ ]:
car={
    "brand":"BM",
    "model":"GT",
    "year" :1988
}
for x in car.values():
  print(x)

BM
GT
1988


In [ ]:
car={
    "brand":"BM",
    "model":"GT",
    "year" :1988
}
for x,y in car.items():
  print(x,y)

brand BM
model GT
year 1988


In [ ]:
car={
    "brand":"BM",
    "model":"GT",
    "year" :1988
}
myCar=car.copy()
print(myCar)

{'brand': 'BM', 'model': 'GT', 'year': 1988}


In [ ]:
car={
    "brand":"BM",
    "model":"GT",
    "year" :1988
}
myCar=dict(car)
print(myCar)

{'brand': 'BM', 'model': 'GT', 'year': 1988}


In [ ]:
name1={"name":"놀부","year":2001}
name2={"name":"흥부","year":2010}
name3={"name":"제비","year":2015}
story={
    "name1":name1,
    "name2":name2,
    "name3":name3
}
print(story)

{'name1': {'name': '놀부', 'year': 2001}, 'name2': {'name': '흥부', 'year': 2010}, 'name3': {'name': '제비', 'year': 2015}}


#CH4 조건문, 반복문  


##중첩 if문

예시) 시험출제 가능성 있음  
"%5d" ->   123  
"%05d"-> 00123

In [ ]:
score=int(input("점수를 입력해 주세요 : "))
if(score > 100):
  print("0~100까지의 숫자로 입력하세요.")
else:
  if (score >= 90):
    print("점수: %3d A"%score)
  elif(score >= 80):
    print("점수: %3d B"%score)
  elif(score >= 70):
    print("점수: %3d C"%score)
  elif(score >= 60):
    print("점수: %3d D"%score)
  else:
    print("점수: %3d F"%score)
print("학점입니다.")

점수를 입력해 주세요 : 96
점수:  96 A
학점입니다.


In [ ]:
import random

numbers=[]
for num in range(0,10):
  numbers.append(random.randrange(0,10))

print("생성된 리스트", numbers)

for num in range(0,10):
  if num not in numbers:
    print("숫자 %d는 리스트에 없습니다."%num)
  else:
    print("숫자 %d는 리스트에 있습니다."%num)

생성된 리스트 [6, 7, 7, 8, 1, 1, 4, 7, 5, 7]
숫자 0는 리스트에 없습니다.
숫자 1는 리스트에 있습니다.
숫자 2는 리스트에 없습니다.
숫자 3는 리스트에 없습니다.
숫자 4는 리스트에 있습니다.
숫자 5는 리스트에 있습니다.
숫자 6는 리스트에 있습니다.
숫자 7는 리스트에 있습니다.
숫자 8는 리스트에 있습니다.
숫자 9는 리스트에 없습니다.


In [ ]:
select, answer, numStr, num1, num2 = 0,0,"",0,0
select=int(input("1.입력한 수식 계산 2.두수의 합계 "))
if select == 1:
  numStr=input("*** 수식을 입력하세요 : ")
  answer=eval(numStr) # 1+2, "string"+"str"
  print("%s 결과는 %5.1f입니다."%(numStr,answer))
elif select==2:
  num1=int(input("*** 첫번째 숫자를 입력하세요 : "))
  num2=int(input("*** 두번째 숫자를 입력하세요 : "))
  for i in range(num1, num2+1):
    answer=answer+i
  print("%d+...+%d는 %d입니다."%(num1,num2,answer))
else:
  print("1또는 2만 입력해야합니다.")

1.입력한 수식 계산 2.두수의 합계 2
*** 첫번째 숫자를 입력하세요 : 5
*** 두번째 숫자를 입력하세요 : 6
5+...+6는 11입니다.


##for & while 반복문

for 변수(i) in (시작값, 끝값, 증가값)  
시험 출제 가능성 있을 듯?

In [ ]:
for i in range(0,3,1):
  print("안녕!")

안녕!
안녕!
안녕!


In [ ]:
for i in [0,3,1]:
  print("안녕!")

안녕!
안녕!
안녕!


In [ ]:
for i in range(0,3,1):
  for j in range(0,5,1):
    print("i:%d, j:%d"%(i,j))

i:0, j:0
i:0, j:1
i:0, j:2
i:0, j:3
i:0, j:4
i:1, j:0
i:1, j:1
i:1, j:2
i:1, j:3
i:1, j:4
i:2, j:0
i:2, j:1
i:2, j:2
i:2, j:3
i:2, j:4


In [ ]:
i,j=0,0
for i in range(1,10,1):
  for j in range(1,10,1):
    print("%d x %d = %2d" %(i,j,i*j), end=' ')
  print("")

1 x 1 =  1  1 x 2 =  2  1 x 3 =  3  1 x 4 =  4  1 x 5 =  5  1 x 6 =  6  1 x 7 =  7  1 x 8 =  8  1 x 9 =  9  
2 x 1 =  2  2 x 2 =  4  2 x 3 =  6  2 x 4 =  8  2 x 5 = 10  2 x 6 = 12  2 x 7 = 14  2 x 8 = 16  2 x 9 = 18  
3 x 1 =  3  3 x 2 =  6  3 x 3 =  9  3 x 4 = 12  3 x 5 = 15  3 x 6 = 18  3 x 7 = 21  3 x 8 = 24  3 x 9 = 27  
4 x 1 =  4  4 x 2 =  8  4 x 3 = 12  4 x 4 = 16  4 x 5 = 20  4 x 6 = 24  4 x 7 = 28  4 x 8 = 32  4 x 9 = 36  
5 x 1 =  5  5 x 2 = 10  5 x 3 = 15  5 x 4 = 20  5 x 5 = 25  5 x 6 = 30  5 x 7 = 35  5 x 8 = 40  5 x 9 = 45  
6 x 1 =  6  6 x 2 = 12  6 x 3 = 18  6 x 4 = 24  6 x 5 = 30  6 x 6 = 36  6 x 7 = 42  6 x 8 = 48  6 x 9 = 54  
7 x 1 =  7  7 x 2 = 14  7 x 3 = 21  7 x 4 = 28  7 x 5 = 35  7 x 6 = 42  7 x 7 = 49  7 x 8 = 56  7 x 9 = 63  
8 x 1 =  8  8 x 2 = 16  8 x 3 = 24  8 x 4 = 32  8 x 5 = 40  8 x 6 = 48  8 x 7 = 56  8 x 8 = 64  8 x 9 = 72  
9 x 1 =  9  9 x 2 = 18  9 x 3 = 27  9 x 4 = 36  9 x 5 = 45  9 x 6 = 54  9 x 7 = 63  9 x 8 = 72  9 x 9 = 81  


In [ ]:
i=0
while i<3:
  print("%d : whlie문" %i)
  i=i+1

0 : whlie문
1 : whlie문
2 : whlie문


In [ ]:
# Q 1부터 10까지 합계를 while문으로
i, hap=0,0
i=1
while i<11:
  hap=hap+i
  i=i+1

print("1부터 10까지의 합계: %d"%hap)

1부터 10까지의 합계: 55


In [ ]:
# Q 1부터 10까지 합계를 for문으로
i, hap=0,0
i=1
for i in range(1,11,1):
  hap=hap+i
print("1부터 10까지의 합계: %d"%hap)

1부터 10까지의 합계: 55


In [ ]:
#while True:
print("무한 루프",end=' ')

무한 루프 

In [ ]:
# break (루프안에서 1번 실행하고 빠짐)
for i in range(1,100):
  print("for문 %d번 실행"%i)
  break
print("break 빠짐")

for문 1번 실행
break 빠짐


In [ ]:
hap=0
a,b=0,0
while True:
  a= int(input("덧셈 첫번째 수 입력 : "))
  if a==0:
    break
  b= int(input("덧셈 두번째 수 입력 : "))
  hap=a+b
  print("%d+%d=%d"%(a,b,hap))
print("break로 탈출")

덧셈 첫번째 수 입력 : 1
덧셈 두번째 수 입력 : 2
1+2=3
덧셈 첫번째 수 입력 : 0
break로 탈출


시험출제 continue

In [ ]:
#continue
hap, i=0,0
for i in range(1,11):
  if i % 2 == 0:
    print("i : %d"%i)
    continue
  hap+=i
print("1~10 합계(3의 배수 제외): %d"%hap)

i : 2
i : 4
i : 6
i : 8
i : 10
1~10 합계(3의 배수 제외): 25


In [ ]:
#continue
hap, i=0,0
for i in range(1,101):
  if i % 3 == 0:
    continue
  hap+=i
print("1~100 합계(3의 배수 제외): %d"%hap)

1~100 합계(3의 배수 제외): 3367


In [ ]:
fruits=["사과","바나나","포도"]
for x in fruits:
  if x == "바나나":
    continue
  print(x)

사과
포도


In [ ]:
for x in range(6):
  print(x)
else:
  print("끝")

0
1
2
3
4
5
끝


In [ ]:
for x in range(6):
  if x ==3: 
    break
  print(x)
else:
  print("끝")

0
1
2


In [ ]:
# for문 루프는 비워둘 수 없음
for x in [0,1,2]:
  pass

## 함수

In [ ]:
# 두 수를 더하는 함수
def plus(v1,v2):
  result=0
  result=v1+v2
  return result

hap=0
hap=plus(10,20)
print("10+20 plus() 처리 결과는 %d" %hap)
hap=0
hap=plus(50,70)
print("50+70 plus() 처리 결과는 %d" %hap)

10+20 plus() 처리 결과는 30
50+70 plus() 처리 결과는 120


In [ ]:
#계산기 코드
def calc(v1,v2,op):
  result=0
  if op=='+':
    result=v1+v2
  elif op == '-':
    result=v1-v2
  elif op == '*':
    result=v1*v2
  elif op == '/':
    result=v1/v2
  return result

rst=0
var1,var2,opr=0,0,""
opr=input("(+,-,*,/) 입력 : ")
var1=int(input("첫번째 수 입력 : "))
var2=int(input("두번째 수 입력 : "))
rst=calc(var1,var2,opr)
print("계산기 %d %s %d = %d"%(var1, opr, var2, rst))

(+,-,*,/) 입력 : *
첫번째 수 입력 : 3
두번째 수 입력 : 2
계산기 3 * 2 = 6


# Q 0으로 나누려고 하면 계산되지 않도록 메시지 출력, 제곱 연산자 출력(숫자1, 연산자, 숫자2) 계산기 함수 만들기

In [ ]:
def calc(v1,v2,op):
  result=0
  if op=='+':
    result=v1+v2
  elif op == '-':
    result=v1-v2
  elif op == '*':
    result=v1*v2
  elif op == '/':
    if v2 == 0:
      print("입력된 값이 0입니다. 다시 입력해주세요.")
    if v1 == 0:
      print("입력된 값이 0입니다. 다시 입력해주세요.")
      continue
    result=v1/v2
  elif op == '**':
    result=v1**v2
  return result

while True:
  opr=input("(+,-,*,/,**) 입력 : ")
  if opr not in ['+', '-', '*', '/', '**']:
    print("잘못된 입력입니다. 다시 입력해주세요.")
    continue
  var1=int(input("첫번째 수 입력 : "))
  var2=int(input("두번째 수 입력 : "))
  rst=calc(var1,var2,opr)
  print("계산기 %d %s %d = %d"%(var1, opr, var2, rst))
  break

(+,-,*,/,**) 입력 : /
첫번째 수 입력 : 0
두번째 수 입력 : 2
입력된 값이 0입니다. 다시 입력해주세요.
계산기 0 / 2 = 0


In [ ]:
def calc(v1,v2,op):
  result=0
  if op=='+':
    result=v1+v2
  elif op == '-':
    result=v1-v2
  elif op == '*':
    result=v1*v2
  elif op == '/':
    result=v1/v2
  elif op == '**':
    result=v1**v2
  return result

rst=0
var1,var2,opr=0,0,""
opr=input("(+,-,*,/,**) 입력 : ")
var1=int(input("첫번째 수 입력 : "))
var2=int(input("두번째 수 입력 : "))
rst=calc(var1,var2,opr)
print("계산기 %d %s %d = %d"%(var1, opr, var2, rst))

(+,-,*,/,**) 입력 : /
첫번째 수 입력 : 2
두번째 수 입력 : 0


KeyboardInterrupt: ignored

In [ ]:
#전역 변수와 지역 변수
def func1():
  a=10 # 지역 변수
  print("func1()에서 값 %d" %a)

def func2():
  print("func2()에서 값 %d" %a)

a=20 #전역 변수

func1()
func2()

func1()에서 값 10
func2()에서 값 20


# Ch5 파일 처리  
"r" read  
"a" append  
"w" wirte  
"x" create  
"t" text  
"b" binary  

In [ ]:
# "w"
from google.colab import files
f=open("a.txt",'w')
f.write("12345")
f.close()

f=open("a.txt",'w')
f.write("abcde")
f.close()

files.download('a.txt')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [ ]:
#'a'
from google.colab import files

f=open("a.txt",'w')
f.write("12345")
f.close()

f=open("a.txt",'a')
f.write("6789")
f.close()

files.download('a.txt')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [ ]:
from google.colab import files
# 생성된 파일이 없을 경우에만 x로 파일 생성 가능
f=open("b.txt",'x')
f.write("abcde")
f.close()

files.download('b.txt')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [ ]:
from google.colab import files
# 생성된 파일이 없을 경우에만 x로 파일 생성 가능, 있으면 에러 발생
f=open("b.txt",'x')
f.write("abcde")
f.close()

files.download('b.txt')

FileExistsError: ignored

In [ ]:
# 여러줄 내용 입력
from google.colab import files
f=open("a.txt",'w')
#f.write("12345678\n987654321\nabcde")
f.write("""1234
4567
890""")
f.close()

files.download('a.txt')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [ ]:
# 리스트, 튜플 내용을 입력
from google.colab import files

t=("1","2","3","4","\n")
l=["a","b","c","d"]
f=open("a.txt",'w')
f.writelines(t)
f.writelines(l)
f.close()

files.download('a.txt')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [ ]:
# 리스트, 튜플 내용을 입력
from google.colab import files

t=("1","2","3","4","\n")
l=["a","b","c","d"]
f=open("a.txt",'w')
for i in range(0,5):
  f.writelines(t)
f.writelines(l)
f.close()

files.download('a.txt')

In [ ]:
#read 모드로 파일 열기
from google.colab import files

f=open("a.txt",'w')
f.write("1234")
f.close()

f=open("a.txt",'r')
print(f.read())
f.close()

files.download('a.txt')

1234


<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [ ]:
# readlin() 함수 : 한줄씩 가져옴
f=open("a.txt",'r')
print(f.readline())
print(f.readline())
f.close()

1234



In [ ]:
from google.colab import files
import os
if os.path.exists("a.txt"):
  os.remove("a.txt") # 폴더를 삭제하려면 os.rmdir("folder")
else:
  print("파일이 존재하지 않음")


In [ ]:
from google.colab import files
import os

print(os.listdir('.'))
os.rename("b.txt","a.txt") # title 이름 변경
print(os.listdir('.'))

['.config', 'b.txt', '.ipynb_checkpoints', 'sample_data']
['.config', 'a.txt', '.ipynb_checkpoints', 'sample_data']


In [ ]:
from google.colab import files
import os
# title 존재 유무 확인
print(os.path.exists('a.txt'))
print(os.path.exists('b.txt'))

True
False


In [ ]:
# 구구단 파일로 저장
from google.colab import files
import sys

f=open("a.txt",'w') # euc-kr
sys.stdout = f
for i in range(2,10):
  for j in range(1,10):
    print("{} x {} = {}". format(i, j, i*j))
  print()
f.close

files.download('a.txt')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [ ]:
f=open("c.txt",'w', encoding='utf8')
f.write("동해물과 백두산이 마르고 닳도록 \n 하느님이 보우하사 \n 우리나라 만세")
f.close()
 
f=open("c.txt",'r', encoding='utf8')
lines=f.readlines()
for line in lines:
  print(line, end='')
f.close



<function TextIOWrapper.close()>

In [ ]:
y=int(input("줄 수 입력하세요 : "))
x=int(y/2)+1

for i in range(1,2*x):
  if(i<=x):
    for j in range(x-i):
      print(" ", end='')
    for j in range(2*i-1):
      print("@", end='')
  print()
else:
  for j in range(i-x):
    print(" ",end='')
  for j in range((2*x-i)*2-1):
    print("@", end='')
  print()

줄 수 입력하세요 : 15


In [ ]:
x=int(input("줄수입력 : "))
for i in range(1,x+1):
  for j in range(x+1-i):
    print("@", end='')
  print()
"""
@@@@@
@@@@
@@@
@@
@
"""

줄수입력 : 15


'\n@@@@@\n@@@@\n@@@\n@@\n@\n'

#전역변수 지정 및 지역 변수 지정

# 타이틀 제작